pkg/authorization/api/v1/types.go

```diff
@@ -476,3 +476,92 @@ type ClusterRoleList struct {
+
+	// Standard object's metadata.
+	kapi.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the matcher.
+	Spec RoleBindingRestrictionSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// RoleBindingRestrictionSpec defines a rolebinding restriction.  Exactly one
+// field must be non-nil.
+type RoleBindingRestrictionSpec struct {
+	// UserRestriction matches against user subjects.
+	UserRestriction *UserRestriction `json:"userrestriction" protobuf:"bytes,1,opt,name=userrestriction"`
+
+	// GroupRestriction matches against group subjects.
+	GroupRestriction *GroupRestriction `json:"grouprestriction" protobuf:"bytes,2,opt,name=grouprestriction"`
+
+	// ServiceAccountRestriction matches against service-account subjects.
+	ServiceAccountRestriction *ServiceAccountRestriction `json:"serviceaccountrestriction" protobuf:"bytes,3,opt,name=serviceaccountrestriction"`
+}
+
+// RoleBindingRestrictionList is a collection of RoleBindingRestriction objects.
+type RoleBindingRestrictionList struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	unversioned.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is a list of RoleBindingRestriction objects.
+	Items []RoleBindingRestriction `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// UserRestriction matches a user either by a string match on the user name,
+// a string match on the name of a group to which the user belongs, or a label
+// selector applied to the user labels.
+type UserRestriction struct {
+	// Users specifies a list of literal user names.
+	Users []string `json:"users" protobuf:"bytes,1,rep,name=users"`
+
+	// Groups specifies a list of literal group names.
+	Groups []string `json:"groups" protobuf:"bytes,2,rep,name=groups"`
+
+	// Selectors specifies a list of label selectors over user labels.
+	Selectors []*unversioned.LabelSelector `json:"labels" protobuf:"bytes,3,rep,name=labels"`
+}
+
+// GroupRestriction matches a group either by a string match on the group name
+// or a label selector applied to group labels.
+type GroupRestriction struct {
+	// Groups is a list of groups used to match against an individual user's
+	// groups. If the user is a member of one of the whitelisted groups, the user
+	// is allowed to be bound to a role.
+	Groups []string `json:"groups" protobuf:"bytes,1,rep,name=groups"`
+
+	// Selectors specifies a list of label selectors over group labels.
+	Selectors []*unversioned.LabelSelector `json:"labels" protobuf:"bytes,2,rep,name=labels"`
+}
+
+// ServiceAccountRestriction matches a service account by a string match on
+// either the service-account name or the name of the service account's
+// namespace.
+type ServiceAccountRestriction struct {
+	// ServiceAccounts specifies a list of literal service-account names.
+	ServiceAccounts []*ServiceAccountReference `json:"serviceaccounts" protobuf:"bytes,1,rep,name=serviceaccounts"`
```

Comment:
I'm not sure, but I was probably following the example of the `Selectors []*unversioned.LabelSelector` fields in `UserRestriction` and `GroupRestriction`, so I will change this if there is no good reason to use pointers for those fields.